src/Service/MoreInfoService/AbstractMoreInfoService.php

```diff
@@ -547,16 +547,18 @@ private function buildSoapResponse(array $searchParameters, array $results): Mor
             $identifierInformation->identifierKnown = true;
             $identifierInformation->coverImage = [];
 
-            $urls = $this->transformer->transformAll($result['imageUrl']);
+            $urls = $this->transformer->transformAll($result['imageUrl'], $result['width'], $result['height']);
             // Unset the original image size as it's not used in soap end-point.
             unset($urls['original']);
             foreach ($urls as $size => $url) {
-                $image = new ImageType();
-                $image->_ = $url;
-                $image->imageSize = $this->imageSizeMapping($size);
-                // @TODO Implement format. Currently all formats return jpeg.
-                $image->imageFormat = $this->getImageFormat('jpeg');
-                $identifierInformation->coverImage[] = $image;
+                if (!is_null($url)) {
```

Comment:
Not sure that we should force this null check on the code calling `transformAll`. It means the that both here and in the `App\Api\Factory\IdentifierFactory` there needs to be logic that relates to transformations and sizes. I would rather that we either add a method that only returns valid transformations.

Alternatively we should send the null URL to the client. This might be a better API design because a client implementation can then always depend on the requested size attributes being present. They will still need to null check the value, but this might be easier/more intuitive then the opposite. @tuj any thoughts on this?

Reply:
API platform throws up if you return NULL... it requires a string

Reply:
Is transformAll used in other places?